src/protobuf/message.cr

```diff
@@ -140,6 +151,9 @@ module Protobuf
           {% end %}
         {% end %}
       )
+        {% for tag, field in FIELDS %}
+          self.{{field[:name].id}} = {{field[:name].id}}
+        {% end %}
```

Comment:
Hmm. Good question! 🤔 

I guess the setter-behavior is only needed for the oneofs
so it could iterate over `ONEOFS` instead of `FIELDS`.

Don't really have a strong opinion there. I guess it would be slightly faster
but could also set a trap for people who subclass to add their own setters
and expect consistent behavior on initialization. 🤔

Reply:
We don't support message inheritance. We implement messages as structs (with the sole exception being recursive messages) to reduce GC pressure and non-abstract structs can't be subclassed in Crystal:

<img width="426" alt="struct Foo followed by struct Bar < Foo throws the compiler error: can't extend non-abstract struct Foo" src="https://user-images.githubusercontent.com/108205/126021693-3f4e2a0c-7074-475d-9b45-d0d9160036d5.png">

The main reason I'm wondering about this is not so much performance but more that setting values in two places gives us two sources of truth. Ideally we'd only set them once. Maybe iterating over `ONEOFS` isn't the right approach in that case, either. I'm not sure.

Reply:
Ah, good point about the inheritance, yes, then that can indeed be ignored.

Hmm yes, I'm not too fond of that part of the PR either but couldn't think of
a better way to do it. But I can't see where a conflict could occur - when you
do `Msg.new` the setters are called, `msg.foo = bar` also uses the setter.

And when deserializing a message it doesn't matter unless your message is
corrupt to begin with (contains multiple values for the same enum).